src/bin/common/solving_loop.ml

```diff
@@ -32,6 +32,8 @@ module DO = D_state_option
 module Sy = Symbols
 module O = Options
 
+exception Exit_on_error of int
```

Comment:
This is also used to exit with a return code of `0`, which is not an error — I'd suggest renaming it to `Exit_with_code`.